test/react-native/features/fixtures/scenario-launcher/scenarios/NetworkRequestCallbackScenario.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect } from 'react'
+import { SafeAreaView, View, Text, StyleSheet } from 'react-native'
+
+export const config = {
+  maximumBatchSize: 1,
+  batchInactivityTimeoutMs: 5000,
+  autoInstrumentAppStarts: false,
+  appVersion: '1.2.3',
+  networkRequestCallback: (networkRequestInfo) => {
+    // don't send the span for the xhr request
+    if (networkRequestInfo.url.indexOf('xhr') > -1) {
+      return null
+    }
+
+    // modify the url for the fetch request
+    return { ...networkRequestInfo, url: networkRequestInfo.url + '&not-your-ordinary-url=true' }
+  }
+}
+
+const fetchSuccess = async () => {
+  try {
+    await fetch('https://google.com/?fetch=true')
```

Comment:
opted for `bugsnag.com` as we don't know the exact maze runner address, leading to having to check against regex strings in the attributes assertion. I was also seeing issues with invalid URIs when trying to use the maze runner address.

Reply:
We don't necessarily know the order of the spans since the other suggested changes - do you think it's better to change it back and use a maze-runner endpoint or is hitting bugsnag.com good enough™?

Reply:
Something like this should do:

```ruby
Then('a span named {string} has an attribute {string} matching the regex {string}') do |span_name, attribute_name, regex|
  spans = spans_from_request_list(Maze::Server.list_for('traces'))
  named_spans = spans.find_all { |span| span['name'] == span_name }

  Maze.check.false(named_spans.empty?)
  Maze.check.true(named_spans.any? do |span|
    span['attributes'].any? do |span_attribute|
      span_attribute['key'] == attribute_name && span_attribute['value']['stringValue'].match?(regex)
    end
  end)
end
```

Avoiding the internet would be good as it's a big source of possible flakes and if we can't reach Maze Runner then the tests would fail anyway

Reply:
Raised PLAT-10956 to cover this change.